src/core/distributions/InverseDistribution.h

```diff
@@ -0,0 +1,175 @@
+     
+    public:
+        // constructor(s)
+        InverseDistribution(const TypedDistribution<valType>& d)
+            : TypedDistribution<valType>( nullptr ),
+              dist( d.clone() )
+        {
+            // add the parameters to our set (in the base class)
+            // in that way other class can easily access the set of our parameters
+            // this will also ensure that the parameters are not getting deleted before we do
+            
+            // add the parameters of the distribution
+            for (const auto& parameter : dist->getParameters()) {
+                // Doesn't just add the parameter, but also links it so that updates cause
+                // the distribution to update
+                this->addParameter( parameter );
+            }
+            
+	        redrawValue();
+        }	
+
+        InverseDistribution(const InverseDistribution &d)
+            : TypedDistribution<valType>( d ),
+              dist( d.dist->clone() )
+        {
+            // add the parameters to our set (in the base class)
+            // in that way other class can easily access the set of our parameters
+            // this will also ensure that the parameters are not getting deleted before we do
+            
+            // add the parameters of the distribution
+            for (const auto& parameter : dist->getParameters()) {
+                this->addParameter( parameter );
+            }
+        }
+
+        // Functions from Distribution.cpp
+
+        // Overriding `touch` was found to be necessary (at 2024-10): 
+        // nodes in the associated tree were not being marked as dirty
+        // when the value of the distribution was changed indirectly.
+        // The override seems not to be necessary for straightforward cases
+        // such as:
+        //     inv_exp ~ dnInv(dnExp(1))
+        //     inv_exp.clamp(42)  # Touched from clamp and from invalidated children
+        // or:
+        //     k ~ dnUniform(1, 10)
+        //     kRates := [ [ 0, k],
+        //                 [ 1, 0] ]
+        //     qK := fnFreeK(kRates)
+        //     invPhy ~ dnInv(dnPhyloCTMC(tree = phylogeny, Q = qK, type = "Standard"))
+        //     invPhy.clamp(chars) # Touched likewise
+        //     k.clamp(1) # Touched likewise
+        // 
+        // The override becomes necessary for subsequent calls to e.g.
+        //     k.clamp(2)
+        //
+        // Further investigation into this latter case may yield a better solution.
+        void touch(const DagNode *affecter, bool touchAll) override
+        {            
+            dist->touch(affecter, touchAll);
+        }
+
+        void restore( const DagNode *restorer )
+        {
+            dist->restore(restorer);
+        }
+
+        void keep( const DagNode* affecter )
+        {
+            dist->keep(affecter);
+        }
+
+        void getAffected(std::set<DagNode *> &affected, DagNode* affecter)
```

Comment:
I think the signature here is supposed to be 
- `RbOrderedSet` instead of `std::set`
- `const DagNode*` instead of `DagNode*`.
If you mark the declarations with `override` it will check that you are indeed overriding a virtual function.

(I ran into this issue while updating `TransformedDistribution`).